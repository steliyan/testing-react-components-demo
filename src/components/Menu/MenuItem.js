import React from "react";

const MenuItem = ({ text }) => {
  return (
    <div>
      {text}
      <button>{text}</button>
    </div>
  );
};

export default MenuItem;
