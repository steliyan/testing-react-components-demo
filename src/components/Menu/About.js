import React from "react";

import MenuItem from "./MenuItem";

export const About = () => {
  return <MenuItem text="About" />;
};
