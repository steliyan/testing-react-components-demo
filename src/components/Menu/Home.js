import React from "react";

import MenuItem from "./MenuItem";

export const Home = () => {
  return <MenuItem text="Home" />;
};
