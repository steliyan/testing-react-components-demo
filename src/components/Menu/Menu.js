import React from "react";

import { Home } from "./Home";
import { About } from "./About";

export const Menu = () => {
  return (
    <div>
      Navigation:
      <Home />
      <About />
    </div>
  );
};
