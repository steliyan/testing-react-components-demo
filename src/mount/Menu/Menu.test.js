import React from "react";
import { mount } from "enzyme";

import { Menu } from "../../components/Menu/Menu";

it("mount render", () => {
  const container = mount(<Menu />);

  expect(container).toMatchSnapshot();
});

it("mount rerender with user", () => {
  const container = mount(<Menu />);

  container.setProps({ user: "Gosho" });

  expect(container).toMatchSnapshot();
});
