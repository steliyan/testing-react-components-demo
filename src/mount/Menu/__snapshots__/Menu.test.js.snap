// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mount render 1`] = `
<Menu>
  <div>
    Navigation:
    <Home>
      <MenuItem
        text="Home"
      >
        <div>
          Home
          <button>
            Home
          </button>
        </div>
      </MenuItem>
    </Home>
    <About>
      <MenuItem
        text="About"
      >
        <div>
          About
          <button>
            About
          </button>
        </div>
      </MenuItem>
    </About>
  </div>
</Menu>
`;

exports[`mount rerender with user 1`] = `
<Menu
  user="Gosho"
>
  <div>
    Hello, Gosho
    Navigation:
    <Home>
      <MenuItem
        text="Home"
      >
        <div>
          Home
          <button>
            Home
          </button>
        </div>
      </MenuItem>
    </Home>
    <About>
      <MenuItem
        text="About"
      >
        <div>
          About
          <button>
            About
          </button>
        </div>
      </MenuItem>
    </About>
  </div>
</Menu>
`;
